frontend/src/heat_map/democracy_viz.js

```diff
@@ -0,0 +1,351 @@
+
+/**
+ * Main component for the map quiz.
+ *
+ * Handles all logic, displays information, and makes database query/posts
+ */
+
+export class DemocracyViz extends  React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            democracyData: null,
+            scoreType: "v2x_polyarchy",
+            year: "1981",
+        };
+    }
+
+    /**
+     * When this component is mounted to the DOM, get democracy score data from the server
+     */
+    async componentDidMount() {
+        try {
+            const res = await fetch('/api/democracy_scores/');
+            const democracy_data = await res.json();
+            this.setState({
+                democracyData: JSON.parse(democracy_data),
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    handleScoreTypeChange (e) {
+        this.setState({
+            scoreType: e.target.value,
+        })
+    }
+
+    handleYearChange (e) {
+        this.setState({
+            year: e.target.value,
+        })
+    }
+
+    render() {
+        if (!this.state.democracyData) {
+            return (<div>Loading!</div>);
+        }
+        return (
+            <>
+                <h1>Democracy in Africa Over Time</h1><hr/>
+                Democracy Score Type: &nbsp;
+                <select onChange={(e) => this.handleScoreTypeChange(e)}>
+                    {Object.keys(this.state.democracyData[0]["democracy_scores"]["1981"]).map(
+                        (entry, i) => (
+                            <option key={i}>
+                                {entry}
+                            </option>
+                        )
+                    )}
+                </select>
+                <br/>
+                Year: &nbsp;
+                <select onChange={(e) => this.handleYearChange(e)}>
+                    {Object.keys(this.state.democracyData[0]["democracy_scores"]).map(
+                        (entry, i) => (
+                            <option key={i}>
+                                {entry}
+                            </option>
+                        )
+                    )}
+                </select>
+                <br/>
+                Currently grey either means:
+                <ul>
+                    {/*Remove this later once we fix data*/}
+                    <li>Country doesn&apos;t exist in CSV</li>
+                    {/*Maybe keep this bottom one*/}
+                    <li>
+                        The country does exist but there is no data for that particular
+                        score type in that particular year
+                    </li>
+                </ul>
+                <br/>
+                <DemocracyMap
+                    democracyData={this.state.democracyData}
+                    scoreType={this.state.scoreType}
+                    year={this.state.year}
+                />
+            </>
+        )
+    }
+}
+
+const scoreTypeRanges = {
```

Comment:
for neatness, please put constants at the top of the file -- this particular constant is fine for prototyping purposes, but let's maybe move this to a data file and read it in the backend. we're going to want it all over the place.